test/com/google/enterprise/adaptor/database/DatabaseAdaptorTest.java

```diff
@@ -1404,6 +1452,45 @@ public void testInitVerifyColumnNames_metadataColumnAliasBad()
     getObjectUnderTest(moreEntries);
   }
 
+  @Test
+  public void testInitVerifyColumnNames_metadataSql_columnInUniqueKey()
+      throws Exception {
+    executeUpdate("create table data(id int, other varchar(20))");
+
+    Map<String, String> moreEntries = new HashMap<String, String>();
+    moreEntries.put("db.modeOfOperation", "rowToText");
+    moreEntries.put("db.uniqueKey", "id:int");
+    moreEntries.put("db.extraMetadataSql",
+        "select other from data where id = ?");
+    moreEntries.put("db.extraMetadataSqlParameters", "id");
```

Comment:
Do you have a test that does not specify this? By default, db.uniqueKey should be used, so we should have a test of that, which could maybe be this one. I think we should have some test where the unique key has two columns, one used in singleDocContentSql, and the other used in extraMetadataSql.